datalad/support/tests/test_annexrepo.py

```diff
@@ -1989,28 +1989,50 @@ def sync_wrapper(push=False, pull=False, commit=False):
 
     # create a subrepo:
     sub = AnnexRepo(opj(ar.path, 'submod'), create=True)
-    # nothing changed, it's empty besides .git, which is ignored
-    eq_(stat, ar.get_status())
 
-    # file in subrepo
-    with open(opj(ar.path, 'submod', 'fourth'), 'w') as f:
-        f.write("this is a birth certificate")
-    stat['untracked'].append(opj('submod', 'fourth'))
-    eq_(stat, ar.get_status())
+    # ATTN: We look at the bundle git rather than "cmd:git" because `git annex
+    # status` will use it regardless of DATALAD_USE_DEFAULT_GIT.
+    bundled = external_versions['cmd:bundled-git']
+    if bundled is external_versions.UNKNOWN:
+        git_version = external_versions['cmd:git']
+    else:
+        git_version = bundled
+    if git_version >= '2.22.0':
+        # Newer Git versions consider a repo without a commit a repository, not
+        # a directory.
+        stat["untracked"].append("submod/")
+        eq_(stat, ar.get_status())
 
-    # add to subrepo
-    sub.add('fourth')
-    sub.commit(msg="birther mod init'ed")
-    stat['untracked'].remove(opj('submod', 'fourth'))
+        # Nothing changes, since the empty repo is still seen as a repo.
+        with open(opj(ar.path, 'submod', 'fourth'), 'w') as f:
```

Comment:
This and some other bits are duplicated with the else block.  Once I had this test passing, I should reworked the code so that shared parts are outside "newer git?" checks.  I'll do that now.